scripts/msmtpq/msmtpq

```diff
@@ -60,16 +60,27 @@ if [[ -z "$NOTIFY_SEND" ]] && [[ ! -t 0 ]] && [[ -n "$DISPLAY" ]] && command -v
 fi
 
 log_later() { LOG_LATER_ARGS=( "$@" ) ; }
-dsp() {
+echo_msg() {
   local L
   local msg=""
   for L; do
     [ -n "$L" ] && msg+="  $L" || msg+="\n"
   done
+  echo "$msg"
+}
+dsp() {
+  msg="$(echo_msg "$@")"
+  echo -e "$msg"
+  if [ "$EMAIL_QUEUE_QUIET" = "0" ] && [ "$NOTIFY_SEND" = 1 ]; then
```

Comment:
Putting the `$EMAIL_QUEUE_QUIET` check here will disable notifications for all messages, even if the user sets `NOTIFY_SEND=1`and just `EMAIL_QUEUE_QUIET=1`.
This specific check should not be inside the generic `dsp()` function, but at the call side. You could for example create an extra function `notify_dsp()`, which calls `dsp()` and displays a notification.
Then at the call side, you could switch between the too with an if condition, that's unfortunately the only way to keep that functionality.
Then for `err()` you just call `notify_dsp()` and `exit 1`afterwards.

Reply:
Calling `scripts/msmtpq/msmtpq --q-mgmt -r <<< ''` shows no desktop notification if the queue is empty. However, it shows a notifcation if there is an error (say invalid login). I thought this is as requested.

> Putting the $EMAIL_QUEUE_QUIET check here will disable notifications for all messages, even if the user sets NOTIFY_SEND=1and just EMAIL_QUEUE_QUIET=1

1. Since you asked for error notifications to be shown under any conditions, `err()` was rewritten so that `dsp()` is no longer used, in particular `err()` making independent of conditions in `dsp()` being met or not.
How come the check for `$EMAIL_QUEUE_QUIET ` in `dsp()` should disable these notifications?

2. `$EMAIL_QUEUE_QUIET=1` was set by default, again as asked for since you wanted to be notified only about the error messages, so why now argue about a scenario that you wanted to avoid?

Reply:
Sorry I think I read the diff wrongly, so for (1) you are correct, I can confirm that this is now finally working as intended, thank you for fixing this. :)

For (2) it is mostly a code-style thing. `dsp()` (read: display) is a generic method, it felt a bit weird to handle specific cases (like quiet email queue) in this generic method, since `dsp()` is also used for other cases that are not related to email queueing (and the `$EMAIL_QUEUE_QUIET` variable should have no influence on that).
However, I looked through all call sides again, and it seems there is no semantical difference right now, so this should be fine and might not be worth the refactoring effort.

The only remaining problem I see, is that some error cases are handled with `err()` now (and thus show a notification), while some other error cases are still handled with `log -e` (which will not show a notification). A further cleanup patch could go through all cases and make this more consistent, but for my purposes this PR addresses my concerns, so this looks good to me.

Reply:
I agree that `$EMAIL_QUEUE_QUIET=1` was a quick and dirty abuse; I left it as it is for backwards compatibility and now have `$NOTIFY_SEND_VERBOSE`.
So `$NOTIFY_SEND_VERBOSE` for desktop and terminal output, `NOTIFY_SEND=1` for only desktop error messages and `$EMAIL_QUEUE_QUIET=1`for only error terminal output. `EMAIL_QUEUE` is a misnomer, but we live with it.